### ISSUE TITLE
how to exclude sync of override parameters selectively  in decalritive setup of app creating other apps. 

### ISSUE THREAD

**Issue by Ankitchandre** [argoproj/argo-cd#4360]
how to exclude sync of override parameters selectively  in decalritive setup of app creating other apps.

**Comment by jessesuen** [comment]
I'm not sure I understand your question. Could you rephrase and provide an example of what you are trying to do?

**Comment by Ankitchandre** [comment]
so when we use Declarative setup for App of Apps as mentioned here (example of guestbook) https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/ and here https://argoproj.github.io/argo-cd/operator-manual/cluster-bootstrapping/

I have a folder with multiple guestbook kind of applications say app1 and app2  .I also create AppProject and namespace using values file in the same folder 

| -- Masterapp
    | -- chart
    | -- templates
         | -- app1.yaml
         | -- app2.yaml
         | -- project.yaml 
         | -- namespace.yaml
    | -- values file
I configure Masterapp in Argo say from test branch . and sync policy automated. 
If i override a value (example secret value) from argo UI in app1 or app2 , The overall status on Masterapp is outofsync. 
When i sync Masterapp , override values are gone from app1 and app2 , and both these apps start failing coz , secret value will change to nil as mentioned in git. 

Any way possible i can keep override in app1 and app2 , when i sync Masterapp ?

**Comment by jessesuen** [comment]
This not possible. The syncing process is nothing but a `kubectl apply` and so the change which you made out-of-band from the kubectl apply will get overridden if it is defined in git. This is the intended behavior of `kubectl apply` and therefore Argo CD.

**Comment by Ankitchandre** [comment]
The parameter is defined in git, i just want to change its value in override params in argo ui and want that not to get synced. Irony is that in some apps it works and in some it doesn't although all apps are defined in  same way as i mentioned above. As you said , if this is not possible then it should not be possible for any of the case.
